OMEdit/OMEditLIB/OMS/OMSSimulationOutputWidget.cpp

```diff
@@ -80,58 +88,176 @@ OMSSimulationOutputWidget::OMSSimulationOutputWidget(const QString &cref, QWidge
   mpArchivedOMSSimulationItem = new ArchivedOMSSimulationItem(mCref, mStartTime, mStopTime, this);
   MainWindow::instance()->getOMSSimulationDialog()->getArchivedSimulationsTreeWidget()->addTopLevelItem(mpArchivedOMSSimulationItem);
   // save the last modified datetime of result file.
-  char *fileName = (char*)"";
+  char *resultFileName = (char*)"";
   int bufferSize;
-  OMSProxy::instance()->getResultFile(mCref, &fileName, &bufferSize);
-  mResultFilePath = QString("%1/%2").arg(OptionsDialog::instance()->getGeneralSettingsPage()->getWorkingDirectory(), QString(fileName));
-  QFileInfo resultFileInfo(mResultFilePath);
-  if (resultFileInfo.exists()) {
-    mResultFileLastModifiedDateTime = resultFileInfo.lastModified();
+  OMSProxy::instance()->getResultFile(mCref, &resultFileName, &bufferSize);
+  mResultFilePath = QString("%1/%2").arg(OptionsDialog::instance()->getGeneralSettingsPage()->getWorkingDirectory(), QString(resultFileName));
+  // save the current datetime as last modified datetime for result file.
+  mResultFileLastModifiedDateTime = QDateTime::currentDateTime();
+  mpSimulationProcess = 0;
+  mIsSimulationProcessKilled = false;
+  mIsSimulationProcessRunning = false;
+  // create subscriber socket
+  mpContext = zmq_ctx_new();
+  mpSubscriberSocket = zmq_socket(mpContext, ZMQ_SUB);
+  int rc = zmq_bind(mpSubscriberSocket, "tcp://127.0.0.1:*");
+  if (rc == 0) {
+    // get the end point
+    const size_t endPointSize = 30;
+    char endPoint[endPointSize];
+    zmq_getsockopt(mpSubscriberSocket, ZMQ_LAST_ENDPOINT, &endPoint, (size_t *)&endPointSize);
+    zmq_setsockopt(mpSubscriberSocket, ZMQ_SUBSCRIBE, "", 0);
+    mpSubscriberSocketTimer = new QTimer;
+    mpSubscriberSocketTimer->setInterval(10);
+    connect(mpSubscriberSocketTimer, SIGNAL(timeout()), SLOT(simulationProgressJson()));
+    mpSubscriberSocketTimer->start();
+    // start the simulation process
+    mpSimulationProcess = new QProcess;
+    mpSimulationProcess->setWorkingDirectory(OptionsDialog::instance()->getGeneralSettingsPage()->getWorkingDirectory());
+    connect(mpSimulationProcess, SIGNAL(started()), SLOT(simulationProcessStarted()));
+    connect(mpSimulationProcess, SIGNAL(readyReadStandardOutput()), SLOT(readSimulationStandardOutput()));
+    connect(mpSimulationProcess, SIGNAL(readyReadStandardError()), SLOT(readSimulationStandardError()));
+  #if (QT_VERSION >= QT_VERSION_CHECK(5, 6, 0))
+    connect(mpSimulationProcess, SIGNAL(errorOccurred(QProcess::ProcessError)), SLOT(simulationProcessError(QProcess::ProcessError)));
+  #else
+    connect(mpSimulationProcess, SIGNAL(error(QProcess::ProcessError)), SLOT(simulationProcessError(QProcess::ProcessError)));
+  #endif
+    connect(mpSimulationProcess, SIGNAL(finished(int,QProcess::ExitStatus)), SLOT(simulationProcessFinished(int,QProcess::ExitStatus)));
+    QStringList args(QString("C:/OpenModelica/OMSimulator/src/OMSimulatorServer/OMSimulatorServer.py"));
```

Comment:
@lochel we need to copy this file to build directory. Right now the path is hard coded.

Reply:
Can you do that?